bot.py

```
@@ -0,0 +1,117 @@
+from telethon import TelegramClient, events, sync
+from telethon.tl.functions.channels import JoinChannelRequest
+import telethon
+from pymongo import MongoClient
+import base64
+import os
+
+print("Input your Telegram API id:")  # Authorisation
+api_id = int(input())
+print("Input your API hash:")
+api_hash = str(input())
+
+resto = TelegramClient('session_name', api_id, api_hash)
+mclient = MongoClient('localhost', 27017)  # MongoDB should be already started
+
+db = mclient.restotexts
+posts = db.posts
+
+channels = {
+            -1001141587803,  # Где вкусно поесть
+            -1001100608817,  # Restoraids
+            -1001164680315,  # FooDiscovery
+            -1001122635434,  # Нерубинштейна
+            -1001348529976,  # ПроКОФьевна
+            -1001244221197,  # ЧИП ДИШ
+            -1001163906641,  # Вкусный Питер
+            -1001379894986,  # Непатрики
+            -1001282209377,   # foodieguide
+            -1001104433974   # Restinmsk
+            }
+
+
+async def main():
+
+    await resto.send_message('me', "start")
+    total = 0
+    e = 0
+
+    for channel in channels:
+        await resto(JoinChannelRequest(channel))
+
+        async for message in resto.iter_messages(channel):
+
+            b64 = None
+
+            if message.photo:
+
+                try:
+                    path = await message.download_media()
+                    b64 = base64.encodebytes(open(path, "rb").read())
+                    os.remove(path)
+
+                except:  # sometimes there is trouble with profile pics
+                    print()
+                    print('=====OOPS!=====')
+                    print(message.chat.title, ": ", message.date)
+                    print('===============')
+                    e += 1
+                    pass

```

`pass` Ничего не делает, может, continue имелся в виду?

---

Там в `except` бесполезное действие для тестов, его нужно убрать. Будет просто `except: pass`